core/src/com/unciv/models/stats/Stats.kt

```diff
@@ -39,11 +39,20 @@ open class Stats() {
 
 
     fun add(stat: Stat, value: Float): Stats {
+        // This looks super inefficient to me, for changing a single stat we assign EVERY stat TWICE
```

Comment:
What  bugs me more is that Stats is an open class with subclasses adding _tons_ of stuff so in those the `isEmpty` or `toString` implementations from Stats are at times really surprising. And surprises lead to bugs.

Anyway, too late, finding an alternative might be a bit tricky - if you want to keep the way these classes are serialized and deserialized intact - and I'd _really_ want to keep _that_ as it is.

What we could _maybe_ do: Ditch the hashMap detour. Leave data members as they are. Add a property to the Stat Enum so you can do a Stat.addTo(target: Stats, value: Float) -> at this point there's actually seven or eight different delegates being called so the 'mapping' comes free...  

Actually,
```kotlin
enum class Stat (val addTo: (target: Stats, value: Float) -> Unit){
    Production ({ target: Stats, value: Float -> target.production += value }),
    Food ({ target: Stats, value: Float -> target.food += value }),
    Gold ({ target: Stats, value: Float -> target.gold += value }),
    Science ({ target: Stats, value: Float -> target.science += value }),
    Culture ({ target: Stats, value: Float -> target.culture += value }),
    Happiness ({ target: Stats, value: Float -> target.happiness += value }),
    Faith  ({ target: Stats, value: Float -> target.faith += value }),
}

// and elsewhere in class Stats
    fun add(stat: Stat, value: Float): Stats {
        stat.addTo(this, value)
        return this
    }
```
compiles just fine, but _testing_ this thoroughly? **S C A R Y** !!!.

Reply:
...no.
If we're doing ANYTHING in that direction, specifying each addition in Stats.add() would be much more readable and much simpler.